common/unsafe/src/test/scala/org/apache/spark/unsafe/types/CollationFactorySuite.scala

```diff
@@ -30,31 +33,101 @@ import org.apache.spark.unsafe.types.UTF8String.{fromString => toUTF8}
 
 class CollationFactorySuite extends AnyFunSuite with Matchers { // scalastyle:ignore funsuite
   test("collationId stability") {
-    val utf8Binary = fetchCollation(0)
+    assert(UTF8_BINARY_COLLATION_ID == 0)
+    assert(INDETERMINATE_COLLATION_ID == 1 << 30)
+
+    val utf8Binary = fetchCollation(UTF8_BINARY_COLLATION_ID)
     assert(utf8Binary.collationName == "UTF8_BINARY")
     assert(utf8Binary.supportsBinaryEquality)
 
-    val utf8BinaryLcase = fetchCollation(1)
+    val utf8BinaryLcase = fetchCollation(UTF8_BINARY_LCASE_COLLATION_ID)
     assert(utf8BinaryLcase.collationName == "UTF8_BINARY_LCASE")
     assert(!utf8BinaryLcase.supportsBinaryEquality)
 
-    val unicode = fetchCollation(2)
+    val unicode = fetchCollation(UNICODE_COLLATION_ID)
     assert(unicode.collationName == "UNICODE")
-    assert(unicode.supportsBinaryEquality);
+    assert(unicode.supportsBinaryEquality)
 
-    val unicodeCi = fetchCollation(3)
+    val unicodeCi = fetchCollation(UNICODE_CI_COLLATION_ID)
     assert(unicodeCi.collationName == "UNICODE_CI")
     assert(!unicodeCi.supportsBinaryEquality)
   }
 
-  test("fetch invalid collation name") {
-    val error = intercept[SparkException] {
-      fetchCollation("UTF8_BS")
+  test("UTF8_BINARY and ICU root locale collation names") {
+    // collation name already normalized
+    Seq(
+      "UTF8_BINARY",
+      "UTF8_BINARY_LCASE",
+      "UTF8_BINARY_UCASE",
+      "UNICODE",
+      "UNICODE_CI",
+      "UNICODE_AI",
+      "UNICODE_CI_AI",
+      "UNICODE_LCASE",
+      "UNICODE_UCASE"
+    ).foreach(collationName => {
+      val col = fetchCollation(collationName)
+      assert(col.collationName == collationName)
+    })
+    // collation name normalization
+    Seq(
+      // UTF8_BINARY
+      ("UTF8_BINARY_CS", "UTF8_BINARY"),
```

Comment:
I don't think that we should allow UTF8_BINARY_CS, UTF8_BINARY_CI and UTF8_BINARY_AS and silently fallback to UTF8_BINARY. IMO, passing UTF8_BINARY_CS should throw an error as invalid collation.

Reply:
Reworked the approach to reflect this. `_AI` and `_CI` will be disabled but `_CS` and `_AS` will be treated as default case and accent sensitive collations with no effect.

Reply:
Approach is different now. We scope this PR to `UTF8_BINARY` and `UTF8_BINARY_LCASE`.